src/ui-kit/experimental/hierarchical/hierarchical-tree-grid/hierarchical-tree-grid.component.ts

```diff
@@ -36,116 +35,149 @@ export interface GridItem {
 })
 export class SamHierarchicalTreeGridComponent implements OnInit {
+
+  /**
+  * Table configurations 
+  */
   @Input() public templateConfigurations: GridTemplateConfiguration;
-  @Input() public template: GridTemplate;
-  @Input() public dataSource: any;
 
-  @Output() public itemSelected = new EventEmitter<GridItem>();
+  /**
+  * Allow to insert a customized template for suggestions to use
+  */
+  @Input() public gridTemplate: GridTemplate;
+
+  /**
+  * Allow to search the data on the table.
+  */
+  @Input() public filterText: string;
+
+    /**
+  * Data for the Table.
+  *  Simple data array
+  * Stream that emit a array each time when the item is selected.
+  * Stream that changes each time when click action trigger on row.
+  */
+  @Input() public gridData: any[] = [];
+
+  /**
+ * Event emitted when level change is clicked
+ */
   @Output() public levelChanged = new EventEmitter<GridItem>();
-  public selectedItem: GridItem;
-  public chkSelected: boolean = false;
-  selectedItemIndex = 0;
 
-  public displayedColumns = ['select'];
-  public samTableDataSource: any | null;
-  dataChange: BehaviorSubject<any[]> = new BehaviorSubject<any[]>([]);
-  public focusedCell: any;
-  public focusedtemIndex = 0;
+  /**
+ * Event emitted when row is clicked
+ */
+  @Output() public rowChanged = new EventEmitter<GridItem>();
 
+  /**
+  * Event emitted when row set is selected.
+  */
+  @Output() selectResults = new EventEmitter<any[]>();
 
-  @ViewChild(SamPaginationComponent) paginator: SamPaginationComponent;
-  @ViewChild(SamSortDirective) sort: SamSortDirective;
-  @ViewChild('filter') filter: ElementRef;
+  public displayedColumns = ['select'];
 
+  public selectedList: any[] = [];
+  public hierarchicalDataSource: HierarchicalDataSource | null;
+  public dataChange: BehaviorSubject<any[]> = new BehaviorSubject<any[]>([]);
+  @ViewChild(SamSortDirective) sort: SamSortDirective;
 
   ngOnChanges() {
-    this.displayedColumns = [...this.displayedColumns, ...this.templateConfigurations.displayedColumns];
+    this.dataChange.next(this.gridData);
+    if (this.hierarchicalDataSource) {
+      this.hierarchicalDataSource.filter = this.filterText;
+    }
   }
 
   ngOnInit() {
-    this.dataChange.next(this.dataSource);
-    this.samTableDataSource = new SampleDataSource(
+    this.displayedColumns = [...this.displayedColumns, ...this.templateConfigurations.displayedColumns];
+  }
+
+  ngAfterViewInit() {
+    this.hierarchicalDataSource = new HierarchicalDataSource(
       this.dataChange,
-      this.paginator,
       this.sort
     );
   }
-
-  onSelectItem(ev: Event, item: GridItem) {
-
-    this.selectedItem = item;
-    this.itemSelected.emit(this.selectedItem);
-
-  }
-  handleKeyup(ev) {
-    if (KeyHelper.is('tab', event)) {
-      return
-    }
-    if (KeyHelper.is('down', event)) {
-      console.log('onDownArrowDown')
+  /**
+   * On select the results
+   */
+  onChecked(ev, row) {
```

Comment:
Can you add type information for both the function parameters and the return type?